### PR TITLE
wasm-size: Compare base..head in size measurements, and apply nightly optimizations.

### PR DESCRIPTION
**Pull request by miguelff**
Follow-up on https://github.com/prisma/prisma-engines/pull/4672
Close https://github.com/prisma/team-orm/issues/869

### CODE REVIEW

File: query-engine/query-engine-wasm/build.sh [hunk 1]
```diff
@@ -28,12 +26,20 @@ then
...
+rustup default nightly-2024-01-25 
+rustup target add wasm32-unknown-unknown
+rustup component add rust-src --target wasm32-unknown-unknown
+
+
 echo "Building query-engine-wasm using $WASM_BUILD_PROFILE profile"
-CARGO_PROFILE_RELEASE_OPT_LEVEL="z" wasm-pack build "--$WASM_BUILD_PROFILE" --target $OUT_TARGET --out-dir "$OUT_FOLDER" --out-name query_engine
+export RUSTFLAGS="-Zlocation-detail=none"
+CARGO_PROFILE_RELEASE_OPT_LEVEL="z" wasm-pack build "--$WASM_BUILD_PROFILE" --target "$OUT_TARGET" --out-dir "$OUT_FOLDER" --out-name query_engine . \
+-Zbuild-std=std,panic_abort -Zbuild-std-features=panic_immediate_abort
```
Comment (aqrln) [inline query-engine/query-engine-wasm/build.sh:38]:
```suggestion
    -Zbuild-std=std,panic_abort -Zbuild-std-features=panic_immediate_abort
```

**Review by aqrln — APPROVED** [review]

File: query-engine/query-engine-wasm/build.sh [hunk 2]
```diff
@@ -28,12 +26,20 @@ then
...
 
+echo "ℹ️  Configuring rust toolchain to use nightly and rust-src component"
+rustup default nightly-2024-01-25 
+rustup target add wasm32-unknown-unknown
+rustup component add rust-src --target wasm32-unknown-unknown
+
+
 echo "Building query-engine-wasm using $WASM_BUILD_PROFILE profile"
-CARGO_PROFILE_RELEASE_OPT_LEVEL="z" wasm-pack build "--$WASM_BUILD_PROFILE" --target $OUT_TARGET --out-dir "$OUT_FOLDER" --out-name query_engine
+export RUSTFLAGS="-Zlocation-detail=none"
```
Comment (jkomyno) [inline query-engine/query-engine-wasm/build.sh:36]: Uhm, does this `RUSTFLAGS` need to be exported?
Couldn't it be defined inline in the `wasm-pack build` command below, like the `-Zbuild-std*` flags?

Comment (aqrln) [reply query-engine/query-engine-wasm/build.sh:36]: it's a compiler flag, not a cargo flag

Comment (aqrln) [reply query-engine/query-engine-wasm/build.sh:36]: we could inline it by calling `RUSTFLAGS="..." wasm-pack ...` as a single command but I think it's fine either way

Comment (jkomyno) [reply query-engine/query-engine-wasm/build.sh:36]: Ok thanks, then I also think this is fine :)

**Review by jkomyno — APPROVED** [review]

**Pull request merged**